### ISSUE TITLE
Bot sit still doing nothing

### ISSUE THREAD

**Issue by GiGaMo24** [joker-ware/JokerCord#3]
All dependencies have been installed. I manage to enter my token, channel id, spam text, time delay. but it forced close after entering the time delay. i change the hash using dhash/phash in the preferences but nothing work either. is there something wrong did i do?

**Comment by joker-ware** [comment]
What did you type in as the time delay? Did you reboot the bot after typing everyting? If nothing works, you can force the setup again by modifying preferences.json and changing "local" to "False"

**Comment by GiGaMo24** [comment]
i type R the first time. after delete preferences.json and copy the json file into the folder again, i type 10. I did reboot and the bot sit there doing nothing. after waiting 30 min still nothing. it doesnt spam txt or catch pokemon after i manually spam the channel.
![nothing](https://user-images.githubusercontent.com/52581921/66241264-3b8cae80-e731-11e9-941f-d82ea56043a4.png)

**Comment by joker-ware** [comment]
Can you please tell me you Python version? You can know this by typing 
```bash
python3 --version
```
in the command line. You can try manually running the script instead of double clicking by typing:
```bash
C:\Python37\python.exe C:/pathtothejoker.py
```
and seeing if that works.

**Comment by GiGaMo24** [comment]
@joker-ware its v3.7
I did try that. its the same thing. it do nothing. i believe it should say 'jokercord is connected' but it didnt. 
I even try running it with IDLE. 
![nothing 2](https://user-images.githubusercontent.com/52581921/66243269-8bba3f80-e736-11e9-9db9-a30942007a3e.png)

**Comment by joker-ware** [comment]
Should not be a huge problem but I have seen Paths with a space in them as yours "\Poke Bot\" give problems and I'm not sure if Python would like that. Can you please remove the space on the Path and try if it works again?

**Comment by GiGaMo24** [comment]
Still doing nothing. 
![nothing 2](https://user-images.githubusercontent.com/52581921/66243677-00da4480-e738-11e9-8193-0b622d2fb7b6.png)

is this for discord app or browser like chrome?

**Comment by joker-ware** [comment]
It should work everywhere, as you are using the token. As long as the token is correct and the channel ID is correct there should be no problems. Just try one last thing, to check whether it is a permissions problem, and instead of double clicking the .py file, try Right Clicking it and running it as admin.

**Comment by GiGaMo24** [comment]
I give up. it still force close on me. It happen after i give admin right to python.exe. if i remove the right, it just sit there and doing nothing. I dont know why the .py cant run as admin on my laptop. I never learn python so im dumb about this. lol

i tried with chrome and the app itself. Im tired right now for another test round. gonna get some sleep first. do tell me if there's any other idea you have @joker-ware . Thanks for helping.

**Comment by joker-ware** [comment]
I'm going to have a look at the code as I have never personally tried it on Windows. I'll tell you when I find something. Thanks for giving it a shot

**Comment by joker-ware** [comment]
Ran it on Windows and managed to get it to work with no effort. Can you please confirm that doing:
```bash
pip3 install discord.py && pip3 install dhash && pip3 install imagehash && pip3 install requests && pip3 install python-resize-image && pip3 install pillow
```
On the console **does** say *requirement already satisfied*? Must have been my fault with the Windows tutorial as I am not very experienced with it. Just paste that an run it, then double click on joker.py and tell me if it works. Remember you can force to start again by changing the preferences.json *"Local"* to **False**.

**Comment by GiGaMo24** [comment]
@joker-ware yeah. all are satisfied
![yep](https://user-images.githubusercontent.com/52581921/66261868-909af400-e807-11e9-93c3-d12b97c60cc8.png)
Already restart it again. Still the same behaviour. Just doing nothing. Im starting to think it was my laptop having problem.

**Comment by joker-ware** [comment]
Sorry to hear about that. You may want to wait until the next update in which I will ad a Web Interface. You can keep up with the project by clicking on "watch" in the main page.

**Comment by joker-ware** [comment]
The new update has been released. Please test it if you can. If no output is give on the terminal, try going to http://localhost:5555 and if you see the webpage you can manage the bot from there.

**Comment by GiGaMo24** [comment]
Aight. @joker-ware . I still got nothing. it didnt do anything at all. T_T
![I try](https://user-images.githubusercontent.com/52581921/66453620-8db63280-ea97-11e9-8dab-f9b07c935ad1.png)
I believe there is nothing much here to see. but this is after reboot after putting the token and channel id. 
So i tried running with IDLE. and i got error? i think. red mean error ig. since the 1st time i ran it without flask module it give me red. hope the pic help.
![i try 2](https://user-images.githubusercontent.com/52581921/66453723-f7ced780-ea97-11e9-8f58-d2c86a4ffa4d.png)

**Comment by joker-ware** [comment]
Alright. It seems like IDLE messes up a bit with discord py. Can you run it manually in the command line or elsewhere?

**Comment by ammasakshay** [comment]
I mean this is how it looks for me too. It doesn't do anything until a pokemon spawns in the server though. (Or you have the spam option set to TRUE) Looks fine?

**Comment by joker-ware** [comment]
If you run it on idle, and have that error, just use normal Python instead. Like on the command line. It seems like IDLE changes the way python behaves and fileno() is an error caused by it

**Comment by GiGaMo24** [comment]
@joker-ware @ammasakshay I did turn on the spam to True. 
and i did run it using double click joker.py and using C:/....Python.exe C:/.....joker.py and still got same result. the bot didn't do anything. 
Do know that i run the terminal with admin right. and python.exe also got admin right.

**Comment by ammasakshay** [comment]
can you screenshot your preferences file (open it in txt) remove whatever you inputted in token and screenshot the rest?
Does it look something like this?
{"hash_type": "phash", "local": "True", "token": "token", "auto_spam": "True", "auto_spam_channel": "channel", "auto_spam_text": "e", "auto_spam_interval": "3", "custom_list": "False"}
with token and channel replaced with the actual token and channel id?

**Comment by GiGaMo24** [comment]
@ammasakshay 
`{"hash_type": "phash", "local": "False", "token": "token", "auto_spam": "True", "auto_spam_channel": "channel", "auto_spam_text": "gogo laaaa", "auto_spam_interval": "7", "custom_list": "False"}`

the local didnt change for me. last version it did.

**Comment by joker-ware** [comment]
Yes, it is now deprecated. The program no longer uses it. I think you are safe removing it

**Comment by joker-ware** [comment]
After you run it manually, is the console printing any type of message?

**Comment by joker-ware** [comment]
If it is a token or channel problem it should say so.

**Comment by GiGaMo24** [comment]
Yep. it saying my token got a problem. I did change it in the jokercord page.

**Comment by joker-ware** [comment]
Can you try typing the token and the channel ID directly in the preferences.json?

**Comment by GiGaMo24** [comment]
@joker-ware after reboot, it still saying my token got problem.

**Comment by joker-ware** [comment]
Just a question. How are you getting your token and, how does it look? Is it a large text? Maybe you are pasting the incorrect thing

**Comment by GiGaMo24** [comment]
using the console, filtering /api then hitting f5. i copied the authorization. i believe it is the same as in the console. it got capital n small letter, number etc. im following your guide tho.

**Comment by GiGaMo24** [comment]
![give up](https://user-images.githubusercontent.com/52581921/66462403-9580d180-eaad-11e9-8e5b-733b14decfc0.png)
this is after 1 hour waiting. you can see that i still got token problem after changing it n reboot. i even use chrome and the discord app to test it. maybe i will try again in the next version. 

closing this until the next version is up.

**Comment by joker-ware** [comment]
I'm going to make a debug version for you, to know exactly what is the error.

**Comment by GiGaMo24** [comment]
ahhh. thanks man. i've seen the debug mode is off in the terminal. thought gonna ask you about it but i forgot. let me know when you want me to test it.

**Comment by GiGaMo24** [comment]
@joker-ware i've tried the new joker.py. but a new problem (or not) came up. I can't produce the token problem. it just like before you release the GUI version. it sit still doing nothing. without anything came up. atleast in the previous it said i got token problem after waiting for 1 hour.

**Comment by joker-ware** [comment]
It definitely is a bot_thread.py problem. I will edit that file so you can run it without the flask server, hoping we can see any type of error. I will try to make everything as verbose as possible.

**Comment by GiGaMo24** [comment]
this is weird tho. ammasakshay can run it while i can't. you also can run the bot without a hiccup. maybe the problem is my own laptop. Idk if the windows update has fuck up anything with the discord thing or other apps since lately they do give problematic update. maybe i need to check thoroughly my laptop if it still failed.

**Comment by joker-ware** [comment]
The only thing I can recommend right now before making the debug version is to reinstall Python from the installer, and making sure to check the add Python to path boxes and everything. That usually fixes problems

**Comment by joker-ware** [comment]
Reinstalling the libraries would also be a good idea

**Comment by GiGaMo24** [comment]
@joker-ware i did reinstall the python as you said. and did checked for add to path option. i dont know how to reinstall library tho. did i need to use `-force-install` command or something?

I already check the laptop and still didn't find anything weird that mess up connection setting. and reading the new issue, what is the odd that discord certificate is the problem for me?

**Comment by joker-ware** [comment]
As far as I have seen, that SSL certificate is not my program's problems, but MacOS Python's. Even of you had that error it should be printed as his. You can force the reinstall of libraries, with:
```bash 
pip3 install discord.py flask requests pillow --force--reinstall
```
Which will just remove the existing versions and install(if any) new versions. Remember this program no longer needs dhash, imagehash and python-resize-image so you can always remove them

**Comment by GiGaMo24** [comment]
@joker-ware alright. i tried the newest update of yours. hoping you could know whats wrong with this error.

![whats next](https://user-images.githubusercontent.com/52581921/66701757-7e064a80-ed32-11e9-8ae4-4df6e6e2d151.png)

**Comment by joker-ware** [comment]
Okay those are good signs. I suppose the bot is spamming correctly? Can you  delete the file /User/guilds.json and download it again from here? If it says key error, there must be a problem with your guilds(that can happen if you downloaded it before I did some changes to the way guilds.json behaved)

**Comment by GiGaMo24** [comment]
nope. sadly it doesnt spam at all. 
I've download and replace guide.json. It still the same. 
No spam either

**Comment by joker-ware** [comment]
Can you send me your guilds.json file? You can censor the names, if you want.

**Comment by GiGaMo24** [comment]
I dont think i need to censor anything. 

![Guild](https://user-images.githubusercontent.com/52581921/66702129-ca538980-ed36-11e9-8364-3577d01dfb40.png)
![guilds](https://user-images.githubusercontent.com/52581921/66702130-ca538980-ed36-11e9-8d2d-b07213f875fb.png)

**Comment by joker-ware** [comment]
Your guilds aren't properly getting written on. Can you again try to run the bot as admin and check if it properly writes the file? There seems to be any type of permissions issue.

**Comment by joker-ware** [comment]
Some user reported failures when writing too, and I suspect the reason. Please wait for the next update, I will push it later

**Comment by FierceMuffin** [comment]
I noticed in the latest screenshot of the bot window GiGaMo24, you got it to stop giving you the error about your token, how did you do that?

**Comment by GiGaMo24** [comment]
ok. you probably need to teach me another way to give permission. I already give python.exe to run as admin. I even try running by command and cmd is running with admin first. but both still give same result. 

@FierceMuffin you just need to reboot the bot in jokercord page after putting it in the preferences

**Comment by FierceMuffin** [comment]
I've done that already, nothing changed. Take a look at my screenshot in issue #10

**Comment by GiGaMo24** [comment]
@FierceMuffin how did you do it? Did you close the bot or you just reboot on the jokercord page after putting the token in the page??

**Comment by FierceMuffin** [comment]
Rebooting from the page just closes the bot, I have to reopen it manually.

**Comment by GiGaMo24** [comment]
well. thats weird. my bot didnt force closed after rebooting. only the jokercord page cant be used since that was the bug. you just need to use localhost:5555 again. dont refresh as that would reboot the bot again

**Comment by FierceMuffin** [comment]
Aye, my bot console just closes altogether, I have to end up restarting the joker.py file then reloading the GUI once joker.py was open again.

**Comment by joker-ware** [comment]
I changed the way JokerCord writes files. Please download joker.py and bot_thread.py again and replace them.

**Comment by joker-ware** [comment]
Please notify my if anything changes

**Comment by FierceMuffin** [comment]
Nothing changed, unfortunately, I input my token and rebooted, had to manually restart the bot console. My token was still in preferences, but the console gave me the same error. The file is updated, it saved my token and channel ID, but for some reason it won't accept the token. Double checked the token in one of my other bots, it still works fine there.

**Comment by joker-ware** [comment]
Is your token a bot token or a normal user token?

**Comment by joker-ware** [comment]
Just to check, it would not be the first time this happens.

**Comment by joker-ware** [comment]
Please update joker.py again, this time when the bot starts it should print to the console your token. If it prints it correctly you should check your token. If not, please tell me.

**Comment by FierceMuffin** [comment]
Alright, updated again, and yes, my token is at the top of the console. However, I still get the same error underneath. The token saved correctly to preferences.json, but for some reason the bot isn't accepting it. I've gone ahead and regenerated the token and used the new one, no change.

**Comment by joker-ware** [comment]
Regenerating? I will repeat my question. Are you getting the token from the applications page, making it a **BOT** token, not an **USER** token? To get your **USER** token you must read my readme in the Usage section

**Comment by joker-ware** [comment]
**BOT** tokens will NOT work with this application.

**Comment by FierceMuffin** [comment]
OH. well there's the problem, sorry, I should not be doing this on no sleep! I thought it wanted the BOT token, my mistake. Wow, I feel dumb. X'D

**Comment by joker-ware** [comment]
No problem man, everyone makes mistakes ;P

**Comment by FierceMuffin** [comment]
I must be doing it again, I followed the instructions for finding my USER token, opened console, navigated to network tab, refreshed, filtered /api, but there is no Applications to click on.

**Comment by joker-ware** [comment]
If that isn't appearing, try clicking on "Library" instead, and copy the Authorization

**Comment by FierceMuffin** [comment]
Got it! It works at last! Such a simple error, I had completely overlooked how to get the token, thinking it was gonna tell me how to get a bot token, which I already knew. Thanks for the help!

**Comment by joker-ware** [comment]
No problem! Glad you solved it!

**Comment by joker-ware** [comment]
@GiGaMo24 Please run this new version as it might solve your problem. Update me with your status

**Comment by FierceMuffin** [comment]
Might be worth adding that detail to the README, specifying that you need a USER token, not a BOT token, just for dummies like me. :)

**Comment by joker-ware** [comment]
Okay, thanks for the suggestion.

**Comment by GiGaMo24** [comment]
@joker-ware I did use the new one. the token printed is the same as mine. but the bot still behave the same error like the last pic.

**Comment by joker-ware** [comment]
How are you running the python file?

**Comment by GiGaMo24** [comment]
the only way i know to run the bot as admin is using cmd. i don't know if giving python.exe admin rights do anything. so i use cmd run in admin, then use the c:path/python.exe c:path/joker.py

**Comment by joker-ware** [comment]
For debugging purposes, can you try:
First change directory 
```bash
cd
```
To the jokercord folder. Then try running it as Python **3** with 
```bash
python3 joker.py
```
Maybe it is running as an older Python?

**Comment by GiGaMo24** [comment]
@joker-ware wait. i only have python.exe. not python3.exe.

**Comment by joker-ware** [comment]
Yes, but try running is as python3 joker.py instead of C:/python.exe C:/joker.py

**Comment by GiGaMo24** [comment]
Nothing happen
![Whats this](https://user-images.githubusercontent.com/52581921/66716042-92167e80-edfc-11e9-9262-bbd4de8ece72.png)

**Comment by joker-ware** [comment]
Are you sure you have Python 3.7 installed? You can try and install it again and remember to add it to the Path

**Comment by GiGaMo24** [comment]
Yep. Im very sure it is python 3.7. its the latest one i downloaded at the page. I'll try reinstall python again

**Comment by GiGaMo24** [comment]
@joker-ware I did try install the 64bit ver python. and im sure i've already added it to the path. it still the same thing. I cant use python3 command on the cmd.

**Comment by joker-ware** [comment]
64 bit python? As I could see in your screenshot your system is 32 bit. You should not be able to install it. Try installing it from here: https://www.python.org/ftp/python/3.7.4/python-3.7.4-embed-win32.zip but before it, I will send you a version that prints your python version to see which one you are executing.

**Comment by GiGaMo24** [comment]
@joker-ware im on 64bit windows. im not using 32bit. but the previous version of python i use is 32bit.

**Comment by joker-ware** [comment]
Please replace bot_thread.py with this file and tell my if it inputs PYTHON VERSION: 3 in top of the console:
[bot_thread.py](http://wikisend.com/download/681250/bot_thread.py)

**Comment by joker-ware** [comment]
Please replace bot_thread.py with this one and tell me if it prints PYTHON VERSION: 3 at the top of the console:
[bot_thread.py](http://wikisend.com/download/275214/bot_thread.py)

**Comment by GiGaMo24** [comment]
ok. im starting to have a headache looking at this
![now what](https://user-images.githubusercontent.com/52581921/66716874-60ee7c00-ee05-11e9-805c-ae35ef0b27d4.png)

**Comment by joker-ware** [comment]
Sorry, somehow you downloaded the old file I sent to you, which had that error xdd

**Comment by joker-ware** [comment]
Can you try the new one?

**Comment by joker-ware** [comment]
It is in the same message, but you read it before I could delete it

**Comment by GiGaMo24** [comment]
Yep. python 3
![now what](https://user-images.githubusercontent.com/52581921/66716968-6a2c1880-ee06-11e9-8138-c1886cecd120.png)

**Comment by joker-ware** [comment]
Lastly, can you try this one?
[bot_thread.py NEW](http://wikisend.com/download/494550/bot_thread.py)

**Comment by GiGaMo24** [comment]
alright. new error came out
![i try 2](https://user-images.githubusercontent.com/52581921/66717100-dfe4b400-ee07-11e9-8b54-b898ca32c67e.png)

**Comment by joker-ware** [comment]
Ahhh... That's what happens when you do things quickly and in a text editor :S
New one: [bot_thread.py Fixed :s](http://wikisend.com/download/313224/bot_thread.py)

**Comment by GiGaMo24** [comment]
uhhh. still the same error. XD

**Comment by joker-ware** [comment]
Gosh darnit. Let me fix it again :(

**Comment by joker-ware** [comment]
Here you are:
[PLEASE WORK](http://wikisend.com/download/403526/bot_thread.py)

**Comment by GiGaMo24** [comment]
It works. But sadly back to old error. :'D
![I try](https://user-images.githubusercontent.com/52581921/66717313-41a61d80-ee0a-11e9-94f4-a5d34ac1e049.png)

**Comment by joker-ware** [comment]
Can you check the contents of /User/guilds.json and tell me if its empty?

**Comment by GiGaMo24** [comment]
yeah. it still empty.
Im starting to think to smash this laptop with a hammer. lol

**Comment by joker-ware** [comment]
The problem seems to be when Joker tries to write to Guilds.json ,that's why it says key error because it's empty. I will send you later a guilds.json file that you will need to fill some data manually to see if it works.

**Comment by joker-ware** [comment]
Could you please replace /User(guilds.json with this one, and change SERVER_ID with the server id you are trying to test in? To get the server ID you just need to enable developer mode and right click the server icon, then copy ID. [/User/guilds.json](http://wikisend.com/download/282102/guilds.json)

**Comment by joker-ware** [comment]
Just for curiosity, does any of your servers have any type of special character like chinese symbols, russian characters, non-english characters...

**Comment by joker-ware** [comment]
Hello, whats your status?

**Comment by GiGaMo24** [comment]
@joker-ware sorry for the late reply. just got back from hiking yesterday. 
this is after i used the enter the the server id in guild.json
![Guild](https://user-images.githubusercontent.com/52581921/66913511-5a4f4700-f047-11e9-941b-5dd722de7102.png)

**Comment by joker-ware** [comment]
Can you send me a screenshot of your guilds.json?

**Comment by GiGaMo24** [comment]
@joker-ware here
![guilds](https://user-images.githubusercontent.com/52581921/66920213-6c37e680-f055-11e9-93c0-f29007d19c9b.png)

**Comment by joker-ware** [comment]
It seems like there is a "3" in the very beggining of your guilds.json file, can you try to delete it?

**Comment by GiGaMo24** [comment]
![give up](https://user-images.githubusercontent.com/52581921/66921603-2597bb80-f058-11e9-80ed-360ac461b2ef.png)
Seems it didnt manage to access the token properly ig? And i did copy the user token. not a bot token.

**Comment by joker-ware** [comment]
You are getting the same problems as another user. Are you using a Mac?
#4

**Comment by joker-ware** [comment]
Of course you are not, lol I saw it. Imma  search a bit on Google. It's a problem with the SSL certificates.

**Comment by GiGaMo24** [comment]
Nope. Im using Windows10. I did try to find the cert problem for windows. but didnt have luck. idk if this is the same problem as using Mac.
and heres again the same previous problem after reboot. Theres also occasianally error about token.

![I try](https://user-images.githubusercontent.com/52581921/66922455-ae632700-f059-11e9-94f4-29524a456fee.png)

**Comment by joker-ware** [comment]
Can you access discordapp.com?

**Comment by GiGaMo24** [comment]
I can. Both on chrome and its app.

**Comment by joker-ware** [comment]
Can you try to run the bot again after having visited the page and tell me if you have the same error?

**Comment by GiGaMo24** [comment]
ouh. that solution? it didnt work. the visit discordapp.com before using bot didnt work. I've tried it before this.

**Comment by joker-ware** [comment]
Yes, the key problem can still be showing because you manually typed only one server's ID. Is the server ID that you typed having problems in the console log?

**Comment by joker-ware** [comment]
It's weird, the SSL problem wasn't showing before

**Comment by GiGaMo24** [comment]
> Yes, the key problem can still be showing because you manually typed only one server's ID. Is the server ID that you typed having problems in the console log?

uhh, how to see that? in the network tab?

**Comment by joker-ware** [comment]
No, I mean, you wrote some server id like this: 49298302928272734 , inside the guilds.json. Did that ID appear in the terminal? Where it says KeyError:"somenumber"

**Comment by GiGaMo24** [comment]
the first number is 6 for my guild id. and most keyerror start with 3 and 4. weird

**Comment by joker-ware** [comment]
Nah, not weird, that's what we are looking for. The key errors that appear are the rest of servers that aren't inside guilds.json. As I work on fixing that I need to see if the bot works inside the server which you copied the ID.

**Comment by joker-ware** [comment]
So, can you try to set the spam channel inside that server and see if it works?

**Comment by GiGaMo24** [comment]
> So, can you try to set the spam channel inside that server and see if it works?

If u mean in the server i want to use the bot i did set it already, but it didnt spam at all. the 1st problem i had was the bot didn't do anything.

**Comment by joker-ware** [comment]
That was happening because guilds.json was empty. I don't know why it isn't working now.
